frame/system/src/lib.rs

```diff
@@ -450,6 +448,19 @@ pub mod pallet {
 			storage::unhashed::kill_prefix(&prefix);
 			Ok(().into())
 		}
+
+		/// Make some on-chain remark and emit event.
+		///
+		/// # <weight>
+		/// - `O(1)`
+		/// - 1 event.
+		/// # </weight>
+		#[pallet::weight(T::SystemWeightInfo::remark_with_event(remark.len() as u32))]
+		pub(crate) fn remark_with_event(origin: OriginFor<T>, remark: Vec<u8>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			Self::deposit_event(Event::Remarked(who, remark));
```

Comment:
But is it really a good idea to put the full remark into the event?

Reply:
Do you have a suggestion? Put hash? Limit the remark length?